src/prefect/server/events/clients.py

```diff
@@ -211,7 +216,8 @@ class PrefectServerEventsClient(EventsClient):
     _publisher: messaging.EventPublisher
 
     async def __aenter__(self) -> Self:
-        self._publisher = messaging.create_event_publisher()
+        publisher = messaging.create_event_publisher()
+        self._publisher = await publisher.__aenter__()
```

Comment:
This was the core of the bug, we weren't `__aenter__`ing the underlying publisher.